### DISCUSSION TITLE
String for Serial MIDI (5 Pin Din) Interface?

### DISCUSSION THREAD

**Discussion by nonsuchpro** [tttapa/Control-Surface#1098] [Q&A]
Hey all. 
What string do I add if I want to use serial MIDI via 5 pin din, not USB? Older sketches would let me choose one or the other for me to use. I'm sure most people are interested in making MIDI controllers to use with a DAW via USB these days but, I need to make a few for hardware synths with only 5 pin I/O.

```
// Instantiate a MIDI Interface to use
//USBMIDI_Interface midi;
(String to use here to instantiate Serial 5 pin MIDI)
```

Basically, what do I use here for this library?
Thanks!

**Comment by tttapa** [comment]
See https://tttapa.github.io/Control-Surface/Doxygen/d3/df7/midi-tutorial.html#midi_md-instantiate-a-midi-interface.

In short: use `HardwareSerialMIDI_Interface midi {Serial};` (or `Serial1`).

**Reply by nonsuchpro** [reply]
Thank you! Cheers!

**Comment by delboy1978uk** [comment]
What are `Serial` and `Serial1`? How do I declare the pins for the DIN socket?

**Reply by tttapa** [reply]
`Serial` and `Serial1` refer to the hardware UART that's being used. The pins are usually determined by the hardware, some boards do not allow you to change the pin assignment in software.

https://docs.arduino.cc/language-reference/en/functions/communication/serial/
